src/Adapter/MSTest.TestAdapter/Execution/TestClassInfo.cs

```diff
@@ -395,14 +395,9 @@ public void RunClassInitialize(TestContext testContext)
                         errorMessage,
                         exceptionStackTraceInfo?.ErrorStackTrace);
 
-                    if (classCleanupLifecycle == ClassCleanupBehavior.EndOfClass)
-                    {
-                        var testFailedException = new TestFailedException(ObjectModelUnitTestOutcome.Failed, errorMessage, exceptionStackTraceInfo);
-                        ClassCleanupException = testFailedException;
-                        throw testFailedException;
-                    }
-
-                    return errorMessage;
+                    var testFailedException = new TestFailedException(ObjectModelUnitTestOutcome.Failed, errorMessage, exceptionStackTraceInfo);
+                    ClassCleanupException = testFailedException;
+                    return testFailedException.Message;
```

Comment:
in here it was returning a message instead of throwing in case of `ClassCleanupBehavior.EndOfAssembly` so I made it returning message as `ClassCleanupBehavior.EndOfClass` is the default now